### PR TITLE
[scudo] Add errno description to mmap failure.

### PR DESCRIPTION
**Pull request by cferris1000**
Added unit tests for all of the linux report error functions.

### CODE REVIEW

File: compiler-rt/lib/scudo/standalone/report_linux.cpp [hunk 1]
```diff
@@ -25,9 +25,10 @@ namespace scudo {
 // Fatal internal map() error (potentially OOM related).
 void NORETURN reportMapError(uptr SizeIfOOM) {
   ScopedString Error;
-  Error.append("Scudo ERROR: internal map failure");
+  Error.append("Scudo ERROR: internal map failure (error desc=%s)",
+               strerror(errno));
   if (SizeIfOOM) {
-    Error.append(" (NO MEMORY) requesting %zuKB", SizeIfOOM >> 10);
+    Error.append(" requesting %zuKB", SizeIfOOM >> 10);
   }
```
Comment (ChiaHungDuan) [inline compiler-rt/lib/scudo/standalone/report_linux.cpp:32]: nit: single-line statement can remove the braces

**Review by ChiaHungDuan — APPROVED** [review]

File: compiler-rt/lib/scudo/standalone/report_linux.cpp (hunk 1 above)
Comment (cferris1000) [reply compiler-rt/lib/scudo/standalone/report_linux.cpp]: Done.

**Comment by vitalybuka** [comment]
This bot is broken https://lab.llvm.org/buildbot/#/builders/169/builds/30273

**Comment by cferris1000** [comment]
Fix has been submitted.

**Pull request merged**